### ISSUE TITLE
numpy 1.13.0 doesn't build with Intel compilers

### ISSUE THREAD

**Issue by boegel** [numpy/numpy#9278]
I'm running into the problem below when trying to build `numpy` 1.13.0 with Intel compilers; building earlier numpy versions (1.12.1, 1.11.1, 1.10.4, ...) with this approach works fine.

```
$ python setup.py build --compiler=intel --fcompiler=intelem
building library "npymath" sources
Found executable /path/to/icc
Could not locate executable ecc
Traceback (most recent call last):
  File "setup.py", line 392, in <module>
    setup_package()
  File "setup.py", line 384, in setup_package
    setup(**metadata)
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/core.py", line 169, in setup
    return old_setup(**new_attr)
  File "/user/home/gent/vsc400/vsc40023/eb_phanpyscratch/CO7/haswell-ib/software/Python/2.7.13-intel-2017a/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/user/home/gent/vsc400/vsc40023/eb_phanpyscratch/CO7/haswell-ib/software/Python/2.7.13-intel-2017a/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/user/home/gent/vsc400/vsc40023/eb_phanpyscratch/CO7/haswell-ib/software/Python/2.7.13-intel-2017a/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/command/build.py", line 47, in run
    old_build.run(self)
  File "/user/home/gent/vsc400/vsc40023/eb_phanpyscratch/CO7/haswell-ib/software/Python/2.7.13-intel-2017a/lib/python2.7/distutils/command/build.py", line 127, in run
    self.run_command(cmd_name)
  File "/user/home/gent/vsc400/vsc40023/eb_phanpyscratch/CO7/haswell-ib/software/Python/2.7.13-intel-2017a/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/user/home/gent/vsc400/vsc40023/eb_phanpyscratch/CO7/haswell-ib/software/Python/2.7.13-intel-2017a/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/command/build_src.py", line 148, in run
    self.build_sources()
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/command/build_src.py", line 159, in build_sources
    self.build_library_sources(*libname_info)
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/command/build_src.py", line 294, in build_library_sources
    sources = self.generate_sources(sources, (lib_name, build_info))
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/command/build_src.py", line 377, in generate_sources
    source = func(extension, build_dir)
  File "numpy/core/setup.py", line 672, in get_mathlib_info
    st = config_cmd.try_link('int main(void) { return 0;}')
  File "/user/home/gent/vsc400/vsc40023/eb_phanpyscratch/CO7/haswell-ib/software/Python/2.7.13-intel-2017a/lib/python2.7/distutils/command/config.py", line 248, in try_link
    self._check_compiler()
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/command/config.py", line 39, in _check_compiler
    old_config._check_compiler(self)
  File "/user/home/gent/vsc400/vsc40023/eb_phanpyscratch/CO7/haswell-ib/software/Python/2.7.13-intel-2017a/lib/python2.7/distutils/command/config.py", line 102, in _check_compiler
    dry_run=self.dry_run, force=1)
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/ccompiler.py", line 713, in new_compiler
    compiler = klass(None, dry_run, force)
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/intelccompiler.py", line 21, in __init__
    v = self.get_version()
  File "/tmp/vsc40023/easybuild_build/numpy/1.13.0/intel-2017a-Python-2.7.13/numpy-1.13.0/numpy/distutils/ccompiler.py", line 582, in CCompiler_get_version
    self.find_executables()
AttributeError: IntelCCompiler instance has no attribute 'find_executables'
```

The `Could not locate executable ecc` seems to be a hint, since this message doesn't occur with older numpy versions where building like this does work:

```
...
building library "npymath" sources
Found executable /path/to/icc
customize IntelEM64TFCompiler
Found executable /path/to/ifort
customize IntelEM64TFCompiler using config
C compiler: icc -fPIC -fp-model strict -O3 -fomit-frame-pointer -openmp
...
```

Why is `numpy` looking for `ecc` when it found `icc`, and not continuing to go and find `ifort` like it did before?

A similar problem was reported by @Eric89GXL in #9101, but there the problem was that `icc` could not be found, that's clearly not the problem here.

Seeing this with Python 2.7.13, Intel compilers 2017.1.132, Intel MKL 2017.1.132.

**Comment by charris** [comment]
#8961 is the only change from 1.12.1 and I don't see how that would be related to this.

**Comment by klicperajo** [comment]
#8961 has added exactly the line causing this (`v = self.get_version()`). I'm not sure why this fails, though.

**Comment by charris** [comment]
So an alternate fix to @8961 suggested in #8941 was `-fopenmp`, which would not be version dependent. Might want to try that.

**Comment by acastanedam** [comment]
Hi everybody,
In regard to this change, in numpy 1.13.0 I've  found using f2py and intel compilers I got the following compilation error:
`mpopt = 'openmp' if v and int(v.split('.')[0]) < 15 else 'qopenmp'
AttributeError: LooseVersion instance has no attribute 'split'`
which can be solved changing the above line by:
`mpopt = 'openmp' if v and int(v.version[0]) < 15 else 'qopenmp'`

Hope this info can be useful,
ACM

**Comment by charris** [comment]
Geez, version is a `LooseVersion` instance. Seems kinda useless given the lack of documentation of the class and the variations of version in the wild. That could be fixed by calling `str` on it I suppose.

So there are two problems:
* `LooseVersion` is  not a string
* missing `find_executables`, which only seems to be defined for fortran compilers

**Comment by charris** [comment]
@boegel Looks like "intel" is so generic that the Itanium platform is also being searched, weird.

**Comment by charris** [comment]
Could those affected try #9285 to see if it fixes the problem? It is a against master but the diff should apply against 1.13.0 without problems.

**Comment by Eric89GXL** [comment]
I can't check now, but if nobody beats me to it I'll give it a shot in a couple of weeks.

**Comment by chengxinlun** [comment]
Still bugged. 
```
customize IntelEM64TFCompiler
Found executable /opt/intel/bin/ifort
Traceback (most recent call last):
  File "setup.py", line 393, in <module>
    setup_package()
  File "setup.py", line 385, in setup_package
    setup(**metadata)
  File "/home/cheng/Downloads/numpy/numpy/distutils/core.py", line 169, in setup
    return old_setup(**new_attr)
  File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/home/cheng/Downloads/numpy/numpy/distutils/command/build.py", line 47, in run
    old_build.run(self)
  File "/usr/lib/python2.7/distutils/command/build.py", line 128, in run
    self.run_command(cmd_name)
  File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/home/cheng/Downloads/numpy/numpy/distutils/command/build_src.py", line 148, in run
    self.build_sources()
  File "/home/cheng/Downloads/numpy/numpy/distutils/command/build_src.py", line 159, in build_sources
    self.build_library_sources(*libname_info)
  File "/home/cheng/Downloads/numpy/numpy/distutils/command/build_src.py", line 294, in build_library_sources
    sources = self.generate_sources(sources, (lib_name, build_info))
  File "/home/cheng/Downloads/numpy/numpy/distutils/command/build_src.py", line 377, in generate_sources
    source = func(extension, build_dir)
  File "numpy/core/setup.py", line 672, in get_mathlib_info
    st = config_cmd.try_link('int main(void) { return 0;}')
  File "/usr/lib/python2.7/distutils/command/config.py", line 248, in try_link
    self._check_compiler()
  File "/home/cheng/Downloads/numpy/numpy/distutils/command/config.py", line 83, in _check_compiler
    self.fcompiler.customize(self.distribution)
  File "/home/cheng/Downloads/numpy/numpy/distutils/fcompiler/__init__.py", line 502, in customize
    get_flags('opt', oflags)
  File "/home/cheng/Downloads/numpy/numpy/distutils/fcompiler/__init__.py", line 493, in get_flags
    flags.extend(getattr(self.flag_vars, tag))
  File "/home/cheng/Downloads/numpy/numpy/distutils/environment.py", line 39, in __getattr__
    return self._get_var(name, conf_desc)
  File "/home/cheng/Downloads/numpy/numpy/distutils/environment.py", line 53, in _get_var
    var = self._hook_handler(name, hook)
  File "/home/cheng/Downloads/numpy/numpy/distutils/fcompiler/__init__.py", line 700, in _environment_hook
    return hook()
  File "/home/cheng/Downloads/numpy/numpy/distutils/fcompiler/intel.py", line 126, in get_flags_opt
    mpopt = 'openmp' if v and v < 15 else 'qopenmp'
  File "/usr/lib/python2.7/distutils/version.py", line 296, in __cmp__
    return cmp(self.version, other.version)
AttributeError: 'int' object has no attribute 'version'
```

The problem seems to be in `mpopt = 'openmp' if v and v < 15 else 'qopenmp'`, where a `version` instance is compared with an int...

**Comment by chengxinlun** [comment]
Change the line to 
`mpopt = 'openmp' if v and int(v.version[0]) < 15 else 'qopenmp'`
solves the problem, as @acastanedam has suggested.

**Comment by charris** [comment]
@chengxinlun Well, `get_version` should return a `LooseVersion` instance which does, or did, work with integers. Can you try putting quotes around "15", also check that v is an instance of `LooseVersion`? What python version is this?

**Comment by charris** [comment]
Oops, checked again, needs a string.

**Comment by charris** [comment]
Should be fixed.

**Comment by chengxinlun** [comment]
Tested, it works. Thx.

**Comment by charris** [comment]
@chengxinlun Thanks for testing! I've made one more modification that I'd like to try. Could you also test that?

**Comment by chengxinlun** [comment]
Well, ok then.

**Comment by charris** [comment]
@chengxinlun Did you test the current version?

**Comment by boegel** [comment]
@charris I just tested the fix in #9285, works like a charm with Intel compilers 17.0.1.132 (2017.1.132)

**Comment by charris** [comment]
@boegel Thanks for the update!
